cli/azd/resources/starter/.github/workflows/azd-up.yml

```diff
@@ -0,0 +1,73 @@
+name: Deploy up to Azure
+
+on:
+  workflow_dispatch:
+  push:
+    # Run when commits are pushed to mainline branch (main or master)
+    # Set this to the mainline branch you use
+    branches:
+      - main
+      - master
+
+# Example GitHub Actions workflow to deploy to Azure using azd
```

Comment:
@savannahostrowski One thought here is that including CI workflow definitions as part of the starter template may spiral out when we start introducing more infra providers / possible combinations.

I could see this being moved to a `azd scaffold pipeline` gesture in the future. But short-term this may be the best we can do.

Reply:
I like this - let's keep the pipeline definitions.

Reply:
Ditto. I love all the comments. Thanks for including them.